virttest/ceph.py

```diff
@@ -24,78 +26,104 @@ class CephError(Exception):
 
 @error_context.context_aware
 def rbd_image_create(ceph_monitor, rbd_pool_name, rbd_image_name,
-                     rbd_image_size, force_create=False):
+                     rbd_image_size, force_create=False, ceph_conf=None):
     """
     Create a rbd image.
     :params ceph_monitor: The specified monitor to connect to
     :params rbd_pool_name: The name of rbd pool
     :params rbd_image_name: The name of rbd image
     :params rbd_image_size: The size of rbd image
     :params force_create: Force create the image or not
+    :params ceph_conf: The path to the ceph configuration file
     """
-    if rbd_image_exist(ceph_monitor, rbd_pool_name, rbd_image_name):
+    if rbd_image_exist(ceph_monitor, rbd_pool_name, rbd_image_name, ceph_conf):
         create_image = False
         image_info = rbd_image_info(ceph_monitor, rbd_pool_name,
-                                    rbd_image_name)
+                                    rbd_image_name, ceph_conf)
         try:
             int(rbd_image_size)
             compare_str = rbd_image_size
         except ValueError:
             compare_str = utils_misc.normalize_data_size(rbd_image_size)
+
         if image_info['size'] != compare_str or force_create:
-            rbd_image_rm(ceph_monitor, rbd_pool_name, rbd_image_name)
+            rbd_image_rm(ceph_monitor, rbd_pool_name, rbd_image_name,
+                         ceph_conf)
             create_image = True
+
     if create_image:
-        cmd = "rbd create %s/%s -m %s" % (rbd_pool_name, rbd_image_name,
-                                          ceph_monitor)
+        cmd = "rbd {opts} create {pool}/{image} {size}"
+        c_opt = '-c %s' % ceph_conf if ceph_conf else ''
+        m_opt = '-m %s' % ceph_monitor if ceph_monitor else ''
+        opts = m_opt + ' ' + c_opt
+        size_bytes = int(float(compare_str))
+        size = '--size=%s' % (size_bytes / 1024 / 1024)
```

Comment:
refer to rbd command usage:
rbd [ -c ceph.conf ] [ -m monaddr ] [--cluster cluster-name]
       [ -p | --pool pool ] [ command ... ]
we can see that it's reasonable to add a new ceph_conf(-c ) argument, and it will only be used when user specifies a customized ceph.conf. It remains the same as before when ceph_conf is None.